Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/validation/TaskValidation.java

```diff
@@ -1,30 +1,45 @@
 package pg.autyzm.friendly_plans.validation;
 
-import android.widget.EditText;
 import database.repository.TaskTemplateRepository;
+import pg.autyzm.friendly_plans.R;
 import pg.autyzm.friendly_plans.string_provider.StringsProvider;
 
 public class TaskValidation extends Validation {
 
-    public TaskValidation(StringsProvider stringsProvider, TaskTemplateRepository taskTmplRepo) {
-        super(stringsProvider, taskTmplRepo);
+    public TaskValidation(StringsProvider stringsProvider,
+            TaskTemplateRepository taskTemplateRepository) {
+        super(stringsProvider, taskTemplateRepository);
     }
 
-    public boolean isValid(EditText name, EditText duration) {
-        if (!hasText(name)) {
-            return false;
+    public ValidationResult isNameValid(String name) {
+        ValidationResult validationResult = isNameString(name, true);
+        if (validationResult.getValidationStatus() == ValidationStatus.INVALID) {
+            return validationResult;
         }
-        if (!isNameOk(name, true)) {
-            return false;
+
+        if (taskTemplateRepository.isNameExists(name)) {
+            return new ValidationResult(ValidationStatus.INVALID,
+                    stringProvider.getString(R.string.name_exist_msg));
         }
-        if (!isNumber(duration, true)) {
-            return false;
+
+        return new ValidationResult(ValidationStatus.VALID);
+    }
+
+    public ValidationResult isNameValid(Long taskId, String name) {
+        ValidationResult validationResult = isNameString(name, true);
+        if (validationResult.getValidationStatus() == ValidationStatus.INVALID) {
+            return validationResult;
         }
 
-        if (taskTmplRepo.isNameExists(name.getText().toString())) {
-            showError(name);
-            return false;
+        if (taskTemplateRepository.isNameExists(taskId, name)) {
+            return new ValidationResult(ValidationStatus.INVALID,
+                    stringProvider.getString(R.string.name_exist_msg));
         }
-        return true;
+
+        return new ValidationResult(ValidationStatus.VALID);
+    }
+
+    public ValidationResult isDurationValid(String duration) {
+        return isNumber(duration, true);
     }
```

Comment:
I have some tips:
1) I'd remove boolean required param from isNumber, isNameString. Passing false to this param seems to just return VALID
2) I think we can handle checking "no zero length" condition in a different way. I have two ideas:
a) If we assume that in future might be a field for which we require rule like "is a string name or empty" then we can use separated method (e. g. isRequired) and after it check isNameString/isNumber (in which we don't require to be not empty, just matches regex)
b) If we treat "no zero length" as a part of isNameString/isNumber rule than we can add it to regex. In general, I'd like to remove isValid method.